SparseGrids/tsgRuleWavelet.cpp

```diff
@@ -332,34 +350,41 @@ inline double RuleWavelet::eval_cubic(int point, double x) const{
             // i.e. 7 or 8
             // These wavelets are reflections across the y-axis of 6 & 5, respectively
             x = -x;
+            sgn = -1.0;
             point = 13 - point;
         }
         point -= 5;
-        return interpolate(&data[2][point*num_data_points],x);
+        return (mode == 0 ? 1.0 : sgn) * interpolate<mode>(&data[2][point*num_data_points],x);
```

Comment:
You have a whole bunch of those, you changed only one above. This one still has the extra multiplication.